src/contexts/ApiProvider/ApiProvider.tsx

```diff
@@ -1,45 +1,46 @@
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 import { GustoEmbeddedProvider } from '@gusto/embedded-api/react-query/_context'
 import { GustoEmbeddedCore } from '@gusto/embedded-api/core'
-import { HTTPClient } from '@gusto/embedded-api/lib/http'
+import { SDKHooks as NativeSDKHooks } from '@gusto/embedded-api/hooks/hooks'
 import { useMemo } from 'react'
+import type { SDKHooks } from '@/types/hooks'
 
-export function ApiProvider({
-  url,
-  headers,
-  children,
-}: {
+export interface ApiProviderProps {
   url: string
   headers?: HeadersInit
+  hooks?: SDKHooks
   children: React.ReactNode
-}) {
-  const httpClientWithHeaders = useMemo(
-    () =>
-      new HTTPClient({
-        fetcher: async request => {
-          if (request instanceof Request && headers) {
-            const headersInstance = new Headers(headers)
-            headersInstance.forEach((headerValue, headerName) => {
-              if (headerValue) {
-                request.headers.set(headerName, headerValue)
-              }
-            })
-          }
-
-          return fetch(request)
-        },
-      }),
-    [headers],
-  )
+}
 
-  const gustoClient = useMemo(
-    () =>
-      new GustoEmbeddedCore({
-        serverURL: url,
-        httpClient: httpClientWithHeaders,
-      }),
-    [httpClientWithHeaders, url],
-  )
+export function ApiProvider({ url, headers, hooks, children }: ApiProviderProps) {
+  const gustoClient = useMemo(() => {
+    // Create native SDKHooks instance and register user hooks
+    const sdkHooks = new NativeSDKHooks()
+
+    // Register user hooks with native SDK
+    hooks?.beforeCreateRequest?.forEach(hook => {
+      sdkHooks.registerBeforeCreateRequestHook(hook)
+    })
```

Comment:
Could be `hooks?.beforeCreateRequest?.forEach(sdkHooks.registerBeforeCreateRequestHook)` since we're passing the first arg along.